### PR TITLE
Add validations for time spines

### PR DESCRIPTION
**Pull request by courtneyholcomb**
### Description
Validations for time spines. 
- Enforce basic naming restrictions
- Ensure uniqueness across dimensions, entities, measures, metrics, semantic models, other custom granularities, and standard granularities
- Ensure time granularity options in where filters are valid

Also bumps to a new production version. This new version of DSI will need to be released to mantle before custom calendar is released as a feature.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: tests/validations/test_time_spines.py [hunk 1]
```diff
@@ -182,3 +188,89 @@ def test_dimension_granularity_smaller_than_time_spine() -> None:  # noqa: D
...
+                    custom_granularities=[
+                        PydanticTimeSpineCustomGranularityColumn(name="retail_year"),
+                    ],
+                ),
+                PydanticTimeSpine(
+                    node_relation=PydanticNodeRelation(alias="time_spine2", schema_name="my_fav_schema"),
+                    primary_column=PydanticTimeSpinePrimaryColumn(name="ds", time_granularity=TimeGranularity.DAY),
+                    custom_granularities=[
+                        PydanticTimeSpineCustomGranularityColumn(name="retail_year"),
+                        PydanticTimeSpineCustomGranularityColumn(name="quarter"),  # todo
```
Comment (WilliamDee) [inline tests/validations/test_time_spines.py:236]: Is something blocking this validation? Seems like its not in this PR

**Review by WilliamDee — APPROVED** [review]

File: tests/validations/test_time_spines.py (hunk 1 above)
Comment (courtneyholcomb) [reply tests/validations/test_time_spines.py:236]: whoops I just forgot to remove the comment! It is validated here!

Comment (courtneyholcomb) [reply tests/validations/test_time_spines.py:236]: it's the "names cannot match reserved time granularity keywords" one

Comment (courtneyholcomb) [reply tests/validations/test_time_spines.py]: fixed it!

Comment (WilliamDee) [reply tests/validations/test_time_spines.py]: ooo didn't realize it was already in the code inside `check_valid_name` and that was done prior to this PR

**Pull request merged**